### PR TITLE
feat examples: add rocket dynamic template (shuttle-static-folder) example

### PR DESCRIPTION
**Pull request by jhawkesworth**
This PR adds 2 example projects to the existing rocket examples, one for showing how to serve static files using shuttle-static-folder and Rocket's FileServ and the other to demonstrate how to use Rocket's dynamic templates mechanism (also needs shuttle-static-folder).

Once there are suitable examples I think  https://github.com/shuttle-hq/shuttle/issues/763 can be closed.

### CODE REVIEW

**Comment by jhawkesworth** [comment]
last change just modified a code comment which wasn't actually true.  I think this is done now so ready for review.

File: rocket/dyn_template_hbs/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#[macro_use]
+extern crate rocket;
+// use rocket::fs::{FileServer, relative};
```
Comment (oddgrd) [inline rocket/dyn_template_hbs/src/main.rs:3]:
```suggestion
```

File: rocket/dyn_template_hbs/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+       Note that shuttle does not include Rocket.toml 
+       so merging config is the preferred way to modify any settings
+       that would otherwise be set in Rocket.toml
+    */
+    let template_dir = static_folder.to_str().unwrap();
+    let figment = rocket::Config::figment()
+	.merge(("template_dir", template_dir));
+    let rocket = rocket::custom(figment)
+	// .mount("/", FileServer::from(relative!("templates")))
```
Comment (oddgrd) [inline rocket/dyn_template_hbs/src/main.rs:36]:
```suggestion
```

File: rocket/dyn_template_hbs/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	.mount("/", routes![index, hello])
+        .attach(Template::fairing())
+        ;
```
Comment (oddgrd) [inline rocket/dyn_template_hbs/src/main.rs:39]: It looks like you need to run rustfmt for this project.

**Review by oddgrd — APPROVED** [review]
Thanks @jhawkesworth! :partying_face: This looks good to me, just missing a few commented out lines and a rustfmt run (which reminds me we should set up some basic CI for this repo).

**Comment by jhawkesworth** [comment]
Oh yes, let me sort out the formatting. Thanks for speedy review again.

-------- Original Message --------

**Comment by jhawkesworth** [comment]
@oddgrd I formatted the code but left the commented out use of FileServ (with explanatory comment).  Hope that's ok.  If you don't like it, just go ahead and take L3-4 and L40-41 out of src/main.rs.  But I ran a test just now and rocket won't serve a favicon for example from the location configured as template_dir without FileServ mount.  None of the other rocket examples show how to use FileServ for serving static content, so I think it is a reasonable thing to have an example to demonstrate.

**Comment by oddgrd** [comment]
Okay, that's fine. Thanks again!

**Pull request merged**